src/RPC/PeerInfoMessage.proto

```diff
@@ -0,0 +1,9 @@
+// awesome.proto
```

Comment:
Are you using GRPC for this? Or just protobuf? If we can make use of GRPC, that would be great, as that aligns it with the rest of Matrix OS.

Reply:
yeah I hadn't got around to making use of gRPC, the use of `protobufjs` was more as a `.proto` file loader as `grpc-js` doesn't have that functionality. They recommend the `@grpc/proto-loader` library so I will switch to that :+1:

Reply:
It turns our `@grpc/proto-loader` uses `protobufjs` under the hood. I couldn't find any way to just purely encode an object into a Buffer in `gRPC`, unless I've missed something. From what I understand, `grpc` is not designed for byo transport/byo socket and it doesn't look like it supports UDP or broadcasting to multiple clients as in Multicast Service Discovery.